iceoryx_hoofs/source/concurrent/loffli.cpp

```diff
@@ -55,7 +55,7 @@ bool LoFFLi::pop(Index_t& index) noexcept
     do
     {
         // we are empty if next points to an element with index of Size
-        if (oldHead.indexToNextFreeIndex >= m_size)
+        if (oldHead.indexToNextFreeIndex >= m_size || !m_nextFreeIndex)
```

Comment:
Why is this needed?

Note that the whole index terminology in loffli is kind of inconsistent or broken which makes it hard to follow (not well-defined, as there are at least two levels of indices from what I remember).

Reply:
@MatthiasKillat By change, I ran into an issue in `LoFFLi` when `init()` was not called. This check prevents `pop()` to work on the data if the `RelativePointer` was not initialized.

Reply:
As there is already a todo, I didn't add a further one.

Reply:
My inner chicken fears that we will forget to remove the unnecessary checks. Could you please add a hint to the todo in the header file? Something like "move init code to c'tor, remove unnecessary `!m_nextfreeIndex` check".